go-controller/pkg/util/iptables.go

```diff
@@ -285,6 +287,22 @@ func (f *FakeIPTables) Delete(tableName, chainName string, rulespec ...string) e
 	return nil
 }
 
+func (f *FakeIPTables) Restore(tableName string, rulesMap map[string][][]string) error {
+	f.Lock()
+	defer f.Unlock()
+	table, err := f.getTable(tableName)
+	if err != nil {
+		return err
+	}
+	for chainName, rules := range rulesMap {
+		for _, rule := range rules {
+			chain, _ := table.getChain(chainName)
```

Comment:
return the error here from getchain like other functions?

Reply:
oh I copied the Insert function, which doesn't check the error. I dont think restore should care if the chain doesnt exist. I'll look more into this.

Reply:
looked into it. This cant fail because in the final commit I changed the code to be:
```
	for chainName, rules := range rulesMap {
		(*table)[chainName] = []string{}
		for _, rule := range rules {
			chain, _ := table.getChain(chainName)
			(*table)[chainName] = append([]string{strings.Join(rule, " ")}, chain...)
		}
	}
```


Where the chain is initialized to empty (as restore would do) before the loop runs. So getChain cant fail.